src/benchmarklib/benchmark_sql_executor.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "sql/sql_pipeline.hpp"
+#include "utils/sqlite_wrapper.hpp"
+
+namespace opossum {
+
+class TransactionContext;
+
+// This class provides SQL functionality to BenchmarkItemRunners. See AbstractBenchmarkItemRunner:_execute_item.
+class BenchmarkSQLExecutor {
+ public:
+  // If sqlite_wrapper is set, all SQL queries executed in the benchmark will also be executed using SQLite and then
+  // validated.
+  BenchmarkSQLExecutor(bool enable_jit, const std::shared_ptr<SQLiteWrapper>& sqlite_wrapper,
+                       const std::optional<std::string>& visualize_prefix);
+
+  std::shared_ptr<const Table> execute(const std::string& sql);
```

Comment:
doc? this doesn't only execute, right? what else does it do? what does it return if "sql" contains multiple statements? or is it only designed for "sql" being a single statement

Reply:
From the perspective of someone implementing a new benchmark, that's the key functionality of this class. The verification / visualization is nothing that I should care about when implementing a new benchmark.

Reply:
all fine, but just a short comment about a long the lines of "this triggers everything to be done for a query, from execution to - optionally - verification and visualization"